helixlauncher-core/src/config.rs

```diff
@@ -165,3 +165,24 @@ fn get_base_path() -> PathBuf {
         .or_else(|| env::current_dir().ok())
         .unwrap()
 }
+
+#[cfg(test)]
+mod tests {
+    use anyhow::{Ok, Result};
+
+    use crate::config::Config;
+
+    #[tokio::test]
+    async fn create_config_in_non_existing_dir() -> Result<()> {
+        let dir = tempfile::tempdir()?;
+        let _ = Config::new_with_data_dir("", "", dir.path().join("abc"))?;
```

Comment:
As these parameters are present, they should get proper values. (tbh I don't fully remember why I left them in even for `new_with_data_dir`, but I'm sure I had reasons)